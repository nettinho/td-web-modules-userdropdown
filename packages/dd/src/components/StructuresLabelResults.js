import React from "react";
import PropTypes from "prop-types";
import { compose } from "redux";
import { connect } from "react-redux";
import { injectIntl } from "react-intl";
import { Label } from "semantic-ui-react";

export const StructuresLabelResults = ({
  intl: { formatMessage },
  structureCount
}) => (
  <Label style={{ float: "right", margin: "10px" }}>
    {structureCount}
    <Label.Detail>
      {formatMessage({ id: "structures.retrieved.results" })}
    </Label.Detail>
  </Label>
);

StructuresLabelResults.propTypes = {
  intl: PropTypes.object,
  structureCount: PropTypes.number
};

const mapStateToProps = ({ structureCount }) => ({ structureCount });

export default compose(
  injectIntl,
  connect(mapStateToProps)
)(StructuresLabelResults);
