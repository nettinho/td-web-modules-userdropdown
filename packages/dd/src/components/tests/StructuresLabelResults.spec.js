import React from "react";
import { shallowWithIntl } from "intl-stub";
import { StructuresLabelResults } from "../StructuresLabelResults";

describe("<StructuresLabelResults />", () => {
  const props = { structureCount: 22 };

  it("matches the latest snapshot", () => {
    const wrapper = shallowWithIntl(<StructuresLabelResults {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
