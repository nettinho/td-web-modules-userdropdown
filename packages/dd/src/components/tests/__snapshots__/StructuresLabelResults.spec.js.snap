// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StructuresLabelResults /> matches the latest snapshot 1`] = `
<Label
  style={
    Object {
      "float": "right",
      "margin": "10px",
    }
  }
>
  22
  <LabelDetail>
    structures.retrieved.results
  </LabelDetail>
</Label>
`;
